package/harvester-os/files/usr/sbin/harv-install

```diff
@@ -180,29 +180,49 @@ preload_rke2_images()
       rke2_tmp="$inst_tmp/rke2"
       mkdir -p $rke2_tmp && tar xf $inst_tmp/rke2.linux-amd64.tar.gz -C $rke2_tmp
 
+      echo "Start rke2 and wait for containerd exist..."
       $rke2_tmp/bin/rke2 server &> /rke2.log &
 
+      until [ -f "/var/lib/rancher/rke2/bin/containerd" ] && [ -f "/var/lib/rancher/rke2/agent/etc/containerd/config.toml" ]
+      do
+        sleep 1
+      done
+
+      echo "Stop rke2 and remove tmporary rke2 files..."
+      pkill rke2
+      rm /rke2.log
+      rm /etc/rancher/rke2/rke2.yaml
+      rm -rf /var/lib/rancher/rke2/server
+      rm -rf /var/lib/rancher/rke2/agent/pod-manifests/*
+
       export PATH=/var/lib/rancher/rke2/bin:$PATH
       export CONTAINERD_ADDRESS=/run/k3s/containerd/containerd.sock
 
+      echo "Start containerd..."
+      containerd -c /var/lib/rancher/rke2/agent/etc/containerd/config.toml -a $CONTAINERD_ADDRESS --state /run/k3s/containerd --root /var/lib/rancher/rke2/agent/containerd &> /containerd.log &
+
       # wait for containerd to be ready
       until ctr --connect-timeout 1s version&>/dev/null
       do
         sleep 1
       done
 
+      # load images
+      for i in /var/lib/rancher/rke2/agent/images/*.tar.zst; do
+        echo "Load images from $i"
+        zstd -d $i -o /usr/local/image.tar
+        ctr -n=k8s.io images import /usr/local/image.tar
+        rm /usr/local/image.tar
```

Comment:
If we have to ship with this workaround, lets avoid decompressing the tarballs to disk, as they can be several GB.
```suggestion
        zstd -d $i | ctr -n=k8s.io images import -
```